src/createBaseRouter.js

```diff
@@ -6,9 +6,24 @@ import warning from 'warning';
 
 import { routerShape } from './PropTypes';
 import RedirectException from './RedirectException';
+import createRender from './createRender';
 import resolveRenderArgs from './utils/resolveRenderArgs';
 
-export default function createBaseRouter({ render }) {
+export default function createBaseRouter({
+  render,
+  renderPending,
+  renderReady,
+  renderError,
+}) {
+  // eslint-disable-next-line no-param-reassign
+  render =
```

Comment:
well, this would require updating the README at least

i'm not 100% sure about this. i put `createRender` at the top level because of bundle size concerns, but perhaps those are irrelevant

i think we would be better off exposing better higher-level abstractions, maybe – something like a `createRelayRouter` in found relay and a `createHashRouter` here, perhaps

how do you find yourself using `createFarceRouter` other than for the relay things with a custom resolver?

Reply:
in practice we've never used a lower level router without `createRender`, to do so you'd need to manage/implement ElementsRenderer yourself as well as StaticContainer. I don't think there are any people who need that level of bundle min/maxing esp if we never do it as the main consumers

Reply:
makes sense. update the docs then?

Reply:
the FarceRouter case could be handled via additional router factories but we're using createConnectedRouter to pass in an external redux store in a few places and that sort of auto drops you into the lowest level api.

There should also be a unified store enhancer i think